builds/jenkins-build-sls-app/Jenkinsfile

```diff
@@ -0,0 +1,1062 @@
+			def flag = false
+
+			for (int i = 0; i < validRuntimes.size(); i++) {
+				if (_runtime == validRuntimes[i]) {
+					flag = true
+				}
+			}
+
+			if (!flag) {
+				echo "$flag"
+				error "Runtime given in the configuration is not valid."
+			}
+		}
+	} else {
+		error "Wrong configuration. Key 'providerRuntime' is missing in the configuration"
+	}
+	if (prop.containsKey("providerTimeout")) {
+		if (prop['providerTimeout'] == "") {
+			error "Wrong configuration. Value for Key 'providerTimeout' is missing in the configuration"
+		} else if (Integer.parseInt(prop['providerTimeout']) > 300) { // Should not be a high
+			error "Wrong configuration. Value for Key 'providerTimeout' should be a less than 160"
+		}
+	} else {
+		error "Wrong configuration. Key 'providerTimeout' is missing in the configuration"
+	}
+
+	if (prop.containsKey("region")) {
+		if (prop['region'] == "") {
+			error "Wrong configuration. Value for Key 'region' is missing in the configuration"
+		} else if (prop['region'] != configLoader.AWS.REGION) {
+			error "Wrong configuration. Value for Key 'region' should be ${region}"
+		}
+	} else {
+		error "Wrong configuration. Key 'region' is missing in the configuration"
+	}
+
+	def runtime = prop['providerRuntime']
+	if (runtime.indexOf("java") > -1) {
+
+		if (prop.containsKey("artifact")) {
+			if (prop['artifact'] == "") {
+				error "Wrong configuration. Value for Key 'artifact' is missing in the configuration"
+			}
+		} else {
+			error "Wrong configuration. Key 'artifact' is missing in the configuration"
+		}
+
+		if (prop.containsKey("mainClass")) {
+			if (prop['mainClass'] == "") {
+				error "Wrong configuration. Value for Key 'mainClass' is missing in the configuration"
+			}
+		} else {
+			error "Wrong configuration. Key 'mainClass' is missing in the configuration"
+		}
+	}
+}
+
+def echoServiceInfo(String env) {
+  try {
+    echo "Deployment info:"
+    sh "serverless --stage $env info -v > deploy-info.txt"
+
+    def arn = "unknown"
+    def svc_response = "unknown"
+    def result = readFile('deploy-info.txt').trim()
+    def resultList = result.tokenize("\n")
+
+    for (item in resultList) {
+      if (item.startsWith("HandlerLambdaFunctionQualifiedArn")) {
+        arn = item.trim().substring(35)
+        version = arn.tokenize(':').last()
+        arn = arn.substring(0, arn.length() - version.length() - 1)
+
+        svc_response = "Your service endpoint: " + arn
+      }
+    }
+
+    echo "==============================================================================================="
+    echo svc_response
+    echo "==============================================================================================="
+
+    return svc_response
+
+  } catch (Exception ex) {
+    echo "Error while getting service info: " + ex.getMessage()
+  }
+}
+
+/**
+	Create the subscription filters and loggroup if not existing
+**/
+def createSubscriptionFilters(stackName, region, roleId) {
```

Comment:
Ok, this is little tricky. Since sls apps can have multiple functions, this has to be written in a way that captures list of all functions defined in the sls yml (of the generated cf json) & attach the log-streamer to all their log groups.

Reply:
Removing stackName variable and substituting it with the sequence of lambda functions retrieved from the deployment_descriptor as suggested

Reply:
https://github.com/tmobile/jazz/pull/829/commits/f334469d3950dbf33226188826b8897bbed9309b